docusaurus/website/docusaurus.config.ts

```
@@ -31,6 +31,7 @@ const config: Config = {
   baseUrl: 'developers/plugin-tools/',
   onBrokenLinks: 'throw',
   onBrokenMarkdownLinks: 'warn',
+  onBrokenAnchors: 'throw',
```

Fails now when broken anchors are detected, instead of only warning